[HOLD] Updated tslint to use api based approach.

Blackbaud-BobbyEarl opened the pull request:
Issue: https://github.com/blackbaud/skyux-sdk-builder/issues/52

**Updates**

Also addresses: https://github.com/blackbaud/skyux-sdk-builder/issues/50

Documented: https://github.com/blackbaud/skyux2-docs/pull/332

Blackbaud-SteveBrush commented on cli/test.js:
```diff
@@ -31,12 +31,11 @@ function test(command, argv) {
       setTimeout(() => {
-        logger.error('Process failed due to linting errors:');
-        lintResult.errors.forEach(error => logger.error(error));
+        logger.error('TSLlint errors displayed again for convenience:');
```
"TSLlint"

Blackbaud-BobbyEarl replied:
Fixed.

Blackbaud-SteveBrush commented on cli/utils/ts-linter.js:
```diff
@@ -1,51 +1,62 @@
-  '--exclude',
-  '**/node_modules/**/*.ts'
-];
+const { Configuration, Linter, Formatters } = tslint;
```
Just my personal preference, but do you need the `tslint` variable anywhere? Could this be imported as:
```
const { Configuration, Linter, Formatters } = require('tslint);
```

Blackbaud-BobbyEarl replied:
Adjusted.

Blackbaud-BobbyEarl commented:
Ready for another look @Blackbaud-SteveBrush.  I addressed the comments you had, including the initializing problem (so that you can change `fit`/`fdescribe` during `skyux watch`).

Blackbaud-SteveBrush commented on cli/utils/ts-linter.js:
```diff
@@ -1,51 +1,85 @@
+    formatter: 'stylish'
+  };
+
+  if (argv.fix) {
```
I believe this line is causing the `e2e` build step to fail.

Blackbaud-BobbyEarl commented:
Closing in favor https://github.com/blackbaud/skyux-sdk-builder/pull/57
